config/src/main/java/org/springframework/security/config/web/util/matcher/RequestMatcherBuilders.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.config.web.util.matcher;
```

Comment:
I agree that these classes could live in `spring-security-config` to align with usage of `ApplicationContext` and `ObjectPostProcessor`. However, I wonder if it's worth thinking more about this? It seems like the lines between what's needed for the DSL and the reusable aspects related to `RequestMatcher` (in `spring-security-web`) could get quite blurry.